analyzer/reader.go

```diff
@@ -0,0 +1,97 @@
+package analyzer
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/brimdata/brimcap/ztail"
+	"github.com/brimdata/zed/compiler"
+	"github.com/brimdata/zed/compiler/ast"
+	"github.com/brimdata/zed/driver"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zng"
+	"github.com/brimdata/zed/zson"
+	"go.uber.org/multierr"
+)
+
+type Reader struct {
+	reader  zio.Reader
+	records int64
+	tailers tailers
+}
+
+func NewReader(ctx context.Context, warner zio.Warner, confs ...Config) (*Reader, error) {
+	var tailers tailers
+	var readers []zio.Reader
+	zctx := zson.NewContext()
+	for _, conf := range confs {
+		reader, tailer, err := tailOne(ctx, zctx, conf, warner)
+		if err != nil {
+			tailers.Close()
+			return nil, err
+		}
+		tailers = append(tailers, tailer)
+		readers = append(readers, reader)
+	}
+	return &Reader{
+		reader:  zio.NewCombiner(ctx, readers),
+		tailers: tailers,
+	}, nil
+}
+
+func (h *Reader) Read() (*zng.Record, error) {
+	rec, err := h.reader.Read()
+	if rec != nil {
+		atomic.AddInt64(&h.records, 1)
+	}
+	return rec, err
+}
+
+func (h *Reader) Stop() error        { return h.tailers.Stop() }
+func (h *Reader) Close() (err error) { return h.tailers.Close() }
```

Comment:
Nit: I don't think there's any need to export any of this.

Reply:
Fair